nemo/collections/common/prompts/formatter.py

```diff
@@ -20,22 +20,40 @@
 EOS_SLOT = "|eos|"
 
 
-class Modality(Enum):
+class BaseModalityType:
+    @staticmethod
+    def matches(value: Any) -> bool:
+        raise NotImplementedError
+
+
+class Text(BaseModalityType):
+    """Modality for text values."""
+
+    @staticmethod
+    def matches(value: str) -> bool:
+        return isinstance(value, str)
+
+
+class TextLiteral(BaseModalityType):
+    def __init__(self, *items):
+        self.allowed_values = items
+
+    def matches(self, value: str):
+        if "<|" in value and "|>" in value:
+            return True  # Special token
+        return isinstance(value, str) and value in self.allowed_values
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}(allowed_values={self.allowed_values})"
```

Comment:
```suggestion
        return f"{self.__class__.__name__}{self.allowed_values}"
```

I think terser is better since `TextLiteral("val1", "val2", "val3")` is self-explanatory